synapse/util/iterutils.py

```diff
@@ -46,10 +45,10 @@ def batch_iter(iterable: Iterable[T], size: int) -> Iterator[Tuple[T]]:
     return iter(lambda: tuple(islice(sourceiter, size)), ())
 
 
-ISeq = TypeVar("ISeq", bound=Sequence, covariant=True)
+ISeq = TypeVar("ISeq", bound=Sequence)
 
 
-def chunk_seq(iseq: ISeq, maxlen: int) -> Iterable[ISeq]:
+def chunk_seq(iseq: ISeq, maxlen: int) -> Iterable[Iterable[ISeq]]:
```

Comment:
:confused: this doesn't look right

Reply:
Oh, hmm...yeah it doesn't. I'm not really sure what the `bound` aspect of a `TypeVar` does. I'll have to look deeper into that.

Any particular reason this isn't using something similar to the above where it takes an `Iterable[T]`? (Well I guess it would need to be a `Collection` so you can call `len` on it.)

Reply:
I updated this to not use `ISeq`, which I don't think is necessary.